client/src/shared/planCard/index.jsx

```diff
@@ -7,53 +7,340 @@ import { Grid } from '@material-ui/core';
 import { useTheme } from '@material-ui/core/styles';
 import UnsafeRender from 'components/UnsafeRender';
 import { Text } from 'shared';
+import moment from 'moment'
+import styled from 'styled-components';
+import BookmarkOutlinedIcon from '../../assets/bookmark-outlined-icon.svg';
+import BookmarkFilledIcon from '../../assets/bookmark-filled-icon.svg';
+import DropPinIcon from '../../assets/drop-pin-icon.svg';
+import { openModal } from '../../redux/modal/slice';
+import { isFavoritePlan } from '../../pages/Plan/hooks';
+import { UserSelectors } from '../../redux/selectors';
+import { useDispatch } from 'react-redux';
+import { subscribeUserToPlan, unsubscribeUserToPlan } from '../../pages/Plan/controller';
+
 
 const PlanCard = ({ plan }) => {
 	const theme = useTheme();
+	const tagsWrapperRef = useRef(null);
+	const [tags, setTags] = useState(plan?.tags || []);
+    const [ isFavPlan, setIsFavPlan ] = useState(false);
+	const tagsRef = useRef([...(plan?.tags || []).map(() => createRef()), createRef()]);
+    const dispatch = useDispatch();
+    const { isAuthenticated, user } = UserSelectors();
+
+    useEffect(() => {
```

Comment:
We can store liked plans in the store, then use selectors to get the ids of the liked plans, instead of fetching them